Feature/authentication

noxi333 opened the pull request.

mvegter committed:
Initial commit

dependabot-preview[bot] committed:
Bump @typescript-eslint/parser from 2.16.0 to 2.20.0

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 2.16.0 to 2.20.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v2.20.0/packages/parser)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump eslint-config-prettier from 6.9.0 to 6.10.0

Bumps [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) from 6.9.0 to 6.10.0.
- [Release notes](https://github.com/prettier/eslint-config-prettier/releases)
- [Changelog](https://github.com/prettier/eslint-config-prettier/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-config-prettier/compare/v6.9.0...v6.10.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

mvegter committed:
refactor: moved app.* to app directory

dependabot-preview[bot] committed:
Bump jest and ts-jest

Bumps [jest](https://github.com/facebook/jest) and [ts-jest](https://github.com/kulshekhar/ts-jest). These dependencies needed to be updated together.

Updates `jest` from 24.9.0 to 25.1.0
- [Release notes](https://github.com/facebook/jest/releases)
- [Changelog](https://github.com/facebook/jest/blob/master/CHANGELOG.md)
- [Commits](https://github.com/facebook/jest/compare/v24.9.0...v25.1.0)

Updates `ts-jest` from 24.3.0 to 25.2.1
- [Release notes](https://github.com/kulshekhar/ts-jest/releases)
- [Changelog](https://github.com/kulshekhar/ts-jest/blob/master/CHANGELOG.md)
- [Commits](https://github.com/kulshekhar/ts-jest/compare/v24.3.0...25.2.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump prettier from 1.18.2 to 1.19.1

Bumps [prettier](https://github.com/prettier/prettier) from 1.18.2 to 1.19.1.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/1.18.2...1.19.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

mvegter committed:
refactor: updated .gitignore

mvegter committed:
chore: added 'nestjs' to cSpell

Commit:
Added todo.md

mvegter committed:
chore: typescript.tsdk should use node_modules

mvegter committed:
feat: swagger

Commit:
feat: Added Docker

mvegter committed:
refactor: use exact version instead of range

dependabot-preview[bot] committed:
chore(deps-dev): bump @types/jest from 24.0.25 to 25.1.3

Bumps [@types/jest](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/jest) from 24.0.25 to 25.1.3.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/jest)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
chore(deps-dev): bump eslint-plugin-import from 2.19.1 to 2.20.1

Bumps [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import) from 2.19.1 to 2.20.1.
- [Release notes](https://github.com/benmosher/eslint-plugin-import/releases)
- [Changelog](https://github.com/benmosher/eslint-plugin-import/blob/master/CHANGELOG.md)
- [Commits](https://github.com/benmosher/eslint-plugin-import/compare/v2.19.1...v2.20.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
chore(deps-dev): bump @nestjs/schematics from 6.8.1 to 6.9.4

Bumps [@nestjs/schematics](https://github.com/nestjs/schematics) from 6.8.1 to 6.9.4.
- [Release notes](https://github.com/nestjs/schematics/releases)
- [Changelog](https://github.com/nestjs/schematics/blob/master/.release-it.json)
- [Commits](https://github.com/nestjs/schematics/commits/6.9.4)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
chore(deps-dev): bump ts-node from 8.6.0 to 8.6.2

Bumps [ts-node](https://github.com/TypeStrong/ts-node) from 8.6.0 to 8.6.2.
- [Release notes](https://github.com/TypeStrong/ts-node/releases)
- [Commits](https://github.com/TypeStrong/ts-node/compare/v8.6.0...v8.6.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
chore(deps): bump @nestjs/common from 6.10.14 to 6.11.8

Bumps [@nestjs/common](https://github.com/nestjs/nest) from 6.10.14 to 6.11.8.
- [Release notes](https://github.com/nestjs/nest/releases)
- [Commits](https://github.com/nestjs/nest/compare/v6.10.14...v6.11.8)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
chore(deps-dev): bump eslint from 6.7.2 to 6.8.0

Bumps [eslint](https://github.com/eslint/eslint) from 6.7.2 to 6.8.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v6.7.2...v6.8.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
chore(deps-dev): bump typescript from 3.7.4 to 3.8.2

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 3.7.4 to 3.8.2.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v3.7.4...v3.8.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

mvegter committed:
feat: added basic pub/sub integration

mvegter committed:
feat: added Docker Compose setup

dependabot-preview[bot] committed:
chore(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 2.12.0 to 2.21.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v2.21.0/packages/eslint-plugin)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Commit:
feat: Added passport(-local) to package-lock and jwt

Commit:
feat: Added .gitattributes

Commit:
feat: added Users

Commit:
feat: Added authentication

Commit:
feat: WIP: Added a authenticated route

Commit:
feat: Added jwt guard and strategy

Commit:
feat: Added jwt to authentication module

Commit:
feat: Added profile route

mvegter commented on src/app/app.controller.ts:
```diff
@@ -1,12 +1,30 @@
-import { Controller, Get } from '@nestjs/common';
+import { Controller, Get, Post, Request, UseGuards } from '@nestjs/common';
 import { AppService } from './app.service';
+import { LocalAuthGuard } from 'src/authentication/guards/local-auth.guard';
```
Imports src won't work

mvegter commented on src/authentication/authentication.controller.ts:
```diff
@@ -0,0 +1,6 @@
+import { Controller } from '@nestjs/common';
+
+@Controller('authentication')
+export class AuthenticationController {
```
Login route

mvegter commented on src/authentication/authentication.module.ts:
```diff
@@ -0,0 +1,29 @@
+    PassportModule.register({defaultStrategy: 'jwt'}),
+    JwtModule.register({
+      secret: jwtConstants.secret,
+      signOptions: { expiresIn: '60s' }
```
Constants

mvegter reviewed (changes requested):
Format Le code using: npm run format

dependabot-preview[bot] committed:
chore(deps-dev): bump @types/node from 13.1.6 to 13.7.4

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 13.1.6 to 13.7.4.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
chore(deps-dev): bump @nestjs/testing from 6.10.14 to 6.11.8

Bumps [@nestjs/testing](https://github.com/nestjs/nest) from 6.10.14 to 6.11.8.
- [Release notes](https://github.com/nestjs/nest/releases)
- [Commits](https://github.com/nestjs/nest/compare/v6.10.14...v6.11.8)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
chore(deps): bump rimraf from 3.0.0 to 3.0.2

Bumps [rimraf](https://github.com/isaacs/rimraf) from 3.0.0 to 3.0.2.
- [Release notes](https://github.com/isaacs/rimraf/releases)
- [Changelog](https://github.com/isaacs/rimraf/blob/master/CHANGELOG.md)
- [Commits](https://github.com/isaacs/rimraf/compare/v3.0.0...v3.0.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
chore(deps-dev): bump @typescript-eslint/parser from 2.20.0 to 2.21.0

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 2.20.0 to 2.21.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v2.21.0/packages/parser)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

mvegter committed:
refactor: added title, description and version to constants

mvegter committed:
refactor: index page should return info about the api

mvegter committed:
feat: added Docker development setup

dependabot-preview[bot] committed:
chore(deps-dev): bump @types/node from 13.7.4 to 13.7.5

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 13.7.4 to 13.7.5.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

mvegter committed:
lint

mvegter committed:
refactor: moved local login to authentication controller

mvegter committed:
feat: added Swagger annotations

dependabot-preview[bot] committed:
chore(deps-dev): bump @types/node from 13.7.5 to 13.7.6

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 13.7.5 to 13.7.6.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

mvegter committed:
feat: added EVENT_MODULE_OPTIONS

Commit:
chore: Format

Commit:
Fix: Removed unused imports
